packages/libs/deer/README.md

```diff
@@ -0,0 +1,162 @@
+# deer
+
+`deer` is a backend agnostic deserialization framework, featuring meaningful error
+messages and context (utilizing [`error-stack`](https://crates.io/crates/error-stack)) and
+a fail-slow behavior by default.
+
+## Fail-Slow
+
+Currently available Rust deserializers have mostly been developed with correctness and
+speed in mind. These are universally beneficial optimizations, but in certain cases (such
+as when collecting user-facing validation feedback) there are relatively few options
+available within Rust that allow for extended evaluation beyond a single error. `deer`
+aims to improve this situation by consciously trading off an acceptable degree of speed to
+enable the surfacing of multiple errors.
+
+## Example
+
+A popular example for `deer` are end-user facing APIs, given the following example:
+
+```rust
+#[derive(serde::Deserialize, deer::Deserialize)]
+struct Body {
+    i8: i8,
+    string: String
+}
+
+fn main() {
+    let payload = json!({
+        "i8": 257,
```

Comment:
Maybe, we want to use `u8` and `256`? That's an error, which happens more often by accident.

Reply:
fixed in [e0bfcd6](https://github.com/hashintel/hash/pull/1163/commits/e0bfcd65d688297910bb00fcb76911a517a4804c) and [5cec334](https://github.com/hashintel/hash/pull/1163/commits/5cec334dfd86f05fc9f46de8a6a770d89cf30827)